### PR TITLE
[DYN-3109] Disallow Python Migration Assistant when a Python3 engine is selected

### PR DESCRIPTION
**Pull request by zeusongit**
### Purpose

[DYN-3109](https://jira.autodesk.com/browse/DYN-3109)
Disallow Python Migration Assistant when a Python3 engine is selected
Screen Shot 2020-10-01 at 3 20 02 PM

Kapture 2020-09-30 at 16 20 26
(PS: I don't know why the gif is so slow, was normal before exporting)

### Declarations

Check these if you believe they are true 

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

File: src/Libraries/PythonNodeModels/Properties/Resources.resx [hunk 1]
```diff
@@ -192,7 +192,7 @@
     <value>Use this dropdown to choose the python version/engine to execute your code.</value>
   </data>
   <data name="PythonScriptEditorMigrationAssistantButtonTooltip" xml:space="preserve">
-    <value>Migration Assistant to update your Python 2 scripts to Python 3.</value>
+    <value>Migration Assistant to update your Python 2 scripts to Python 3. Engine must be set to Python 2 version to work.</value>
```
Comment (QilongTang) [inline src/Libraries/PythonNodeModels/Properties/Resources.resx:195]: @Amoursol @aparajit-pratap Any more opinion on the messaging?

**Review by QilongTang — APPROVED** [review]
Look solid to me, using converter seems a good choice here

**Comment by aparajit-pratap** [comment]
@zeusongit could you check the test failures.

File: src/Libraries/PythonNodeModels/Properties/Resources.resx (hunk 1 above)
Comment (aparajit-pratap) [reply src/Libraries/PythonNodeModels/Properties/Resources.resx:195]: Looks good to me.

**Review by aparajit-pratap — COMMENTED** [review]
So is it true that without any hover effect, the only difference between the enabled and disabled state of the 2-3 button is that the tooltip appears and does not appear respectively?

**Comment by zeusongit** [comment]
> So is it true that without any hover effect, the only difference between the enabled and disabled state of the 2-3 button is that the tooltip appears and does not appear respectively?
There is also a slight difference in the border of the button.

**Comment by QilongTang** [comment]
Running this branch on the BRE self CI to confirm the failures are real: https://master-15.jenkins.autodesk.com/view/DYN/job/DYN-DevCI_Self_Service/725/

File: src/DynamoCoreWpf/DynamoCoreWpf.csproj [hunk 2]
```diff
@@ -1248,6 +1248,10 @@
       <Project>{6e0a079e-85f1-45a1-ad5b-9855e4344809}</Project>
       <Name>Units</Name>
     </ProjectReference>
+    <ProjectReference Include="..\Libraries\PythonNodeModels\PythonNodeModels.csproj">
+      <Project>{8872CA17-C10D-43B9-8393-5C5A57065EB0}</Project>
+      <Name>PythonNodeModels</Name>
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/DynamoCoreWpf.csproj:1253]: hmm @zeusongit so now DynamoCoreWPF depends on PythonNodeModels, please add `Private False` to this entry to see if it helps the failing test issue.

File: src/Libraries/PythonNodeModelsWpf/ScriptEditorWindow.xaml.cs [hunk 3]
```diff
@@ -1,5 +1,7 @@
 ﻿using System;
+using System.Globalization;
 using System.Windows;
+using System.Windows.Data;
```
Comment (QilongTang) [inline src/Libraries/PythonNodeModelsWpf/ScriptEditorWindow.xaml.cs:4]: Do we still need these references?

File: src/Libraries/PythonNodeModelsWpf/ValueConverter.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+﻿using PythonNodeModels;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
```
Comment (QilongTang) [inline src/Libraries/PythonNodeModelsWpf/ValueConverter.cs:4]: same question here. Are all of these used?

**Review by QilongTang — APPROVED** [review]
LGTM with a comment about using.

File: src/Libraries/PythonNodeModelsWpf/ValueConverter.cs (hunk 4 above)
Comment (zeusongit) [reply src/Libraries/PythonNodeModelsWpf/ValueConverter.cs:3]: yes these are used, removed the other unused ones.

**Pull request merged**